packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -536,8 +540,11 @@ export abstract class SharedObjectCore<
 		this.emitInternal("pre-op", message, local, this);
 
 		this.opProcessingHelper.measure(
-			() => {
+			(metricsTracker) => {
 				this.processCore(message, local, localOpMetadata);
+				metricsTracker.incrementMetric({
```

Comment:
One thing I don't like about the current design is that the callback we pass to measure needs to do this, and thus the call to `incrementMetric()` is becoming part of what gets measured when we calculate duration, and that might "interfere" with having an accurate idea of how long the actual callback took (especially if we had a scenario where the callback is very fast compared to `incrementMetric()`) . I still haven't been able to put proper thought into this, but feel like we should expect the callback to return the property bag it wants us to accumulate, but the actual accumulation should be part of what `measure()` itself does, instead of asking the callback to do it. Not sure how that interacts with the start/measure/end pattern currently in place, which is part of what I still need to think about :)

Reply:
Okay, found one bug with the `TelemetryEventMetrics` implementation. Returning the new instance of `TelemetryEventMetrics<TCustomMetrics>(defaultMetrics)` makes the `SampledTelemetryHelper` to increment values _over_ the given `defaultMetrics` whenever the `measure()` is called. 

So, if there is a default custom data of `{'foo":1, 'bar': 2}` and we wish to increment values by `i` over the 2-iterations (assume `i` starts at `1`)
- Iteration 1: `{foo: 1 + (1), bar: 2 + (1)}`
- Iteration 2: `{foo: 2 + (1 + 2), bar: 3 + (2 + 2)`